R/ValidateModel.R

```diff
@@ -269,3 +269,42 @@ checkNamesandOrdering <- function(n1, n2, note) {
+printValidationResults <- function(model) {
+  print("Validate that commodity output can be recalculated (within 1%) with the model total requirements matrix (L) and demand vector (y) for US production")
+  econval <- compareOutputandLeontiefXDemand(model, tolerance = 0.01)
+  print(paste("Number of sectors passing:",econval$N_Pass))
+  print(paste("Number of sectors failing:",econval$N_Fail))
+  print(paste("Sectors failing:", paste(unique(econval$Failure$rownames), collapse = ", ")))
+  
+  print("Validate that commodity output can be recalculated (within 1%) with model total domestic requirements matrix (L_d) and model demand (y) for US production")
+  econval <- compareOutputandLeontiefXDemand(model, use_domestic=TRUE, tolerance = 0.01)
+  print(paste("Number of sectors passing:",econval$N_Pass))
+  print(paste("Number of sectors failing:",econval$N_Fail))
+  print(paste("Sectors failing:", paste(unique(econval$Failure$rownames), collapse = ", ")))
+  
+  print("Validate that flow totals by commodity (E_c) can be recalculated (within 1%) using the model satellite matrix (B), market shares matrix (V_n), total requirements matrix (L), and demand vector (y) for US production")
+  modelval <- compareEandLCIResult(model, tolerance = 0.01)
+  print(paste("Number of flow totals by commodity passing:",modelval$N_Pass))
+  print(paste("Number of flow totals by commodity failing:",modelval$N_Fail))
+  
+  print("Validate that flow totals by commodity (E_c) can be recalculated (within 1%) using the model satellite matrix (B), market shares matrix (V_n), total domestic requirements matrix (L_d), and demand vector (y) for US production")
+  dom_val <- compareEandLCIResult(model, use_domestic=TRUE, tolerance = 0.01)
+  print(paste("Number of flow totals by commodity passing:",dom_val$N_Pass))
+  print(paste("Number of flow totals by commodity failing:",dom_val$N_Fail))
+  print(paste("Sectors with flow totals failing:", paste(unique(dom_val$Failure$variable), collapse = ", ")))  
+  
+  print("Validate that commodity output are properly transformed to industry output via MarketShare")
+  q_x_val <- compareCommodityOutputXMarketShareandIndustryOutputwithCPITransformation(model, tolerance = 0.01)
+  print(paste("Number of flow totals by commodity passing:",q_x_val$N_Pass))
+  print(paste("Number of flow totals by commodity failing:",q_x_val$N_Fail))
+  print(paste("Sectors with flow totals failing:", paste(unique(q_x_val$Failure$rownames), collapse = ", ")))
+  
+  print("Validate that commodity output equals to domestic use plus production demand")
+  q_val <- compareCommodityOutputandDomesticUseplusProductionDemand(model, tolerance = 0.01)
+  print(paste("Number of flow totals by commodity passing:",q_val$N_Pass))
+  print(paste("Number of flow totals by commodity failing:",q_val$N_Fail))
+  print(paste("Sectors with flow totals failing:", paste(unique(q_val$Failure$rownames), collapse = ", ")))
```

Comment:
Per our discussion in Issue https://github.com/USEPA/useeior/issues/190, make this validation optional?

```suggestion
if (model$specs$CommodityorIndustryType=="Commodity") {
  print("Validate that commodity output equals to domestic use plus production demand")
  q_val <- compareCommodityOutputandDomesticUseplusProductionDemand(model, tolerance = 0.01)
  print(paste("Number of flow totals by commodity passing:",q_val$N_Pass))
  print(paste("Number of flow totals by commodity failing:",q_val$N_Fail))
  print(paste("Sectors with flow totals failing:", paste(unique(q_val$Failure$rownames), collapse = ", ")))
}
```